src/main/java/service/Operator.java

```diff
@@ -0,0 +1,55 @@
+package main.java.service;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.BiFunction;
+
+public enum Operator {
+    PLUS("+", (a, b) -> (a + b)),
+    MINUS("-", (a, b) -> (a - b)),
+    MULTIPLY("*", (a, b) -> (a * b)),
+    DIVIDE("/", (a, b) -> (a / b));
+
+    public final String optName;
+    private final BiFunction<Integer, Integer, Integer> biFunction;
+
+    Operator(String optName, BiFunction<Integer, Integer, Integer> biFunction) {
+        this.optName = optName;
+        this.biFunction = biFunction;
+    }
+
+    public boolean decideToCalculate() {
+        if(this == PLUS || this == MINUS)
+            return true;
+        return false;
+    }
+
+    public Integer calculateOpt(Integer a, Integer b) {
+        return this.biFunction.apply(a, b);
+    }
+
+    // for checking valid operator input.
+    private static List<String> toStringArray(Operator[] operators) {
+        String[] optNames = new String[operators.length];
+        for(int i = 0; i < operators.length; i++)
+            optNames[i] = operators[i].optName;
+        return Arrays.asList(optNames);
+    }
+
+    public static boolean isValidOperator(String comp) {
+        // Operator 의 optName(+ - / =) 에 해당하는 optNames List를 만들고
+        // 인자인 comp가 optNames안에 들어있는지 검사.
+        List<String> optNames = toStringArray(Operator.values());
+        if(optNames.contains(comp))
+            return true;
+        return false;
+    }
+
+    public static Operator stringToOperator(String operator) {
+        return Arrays.stream(values())
+                .filter(value -> value.optName.equals(operator))
+                .findAny()
+                .orElse(null);
```

Comment:
```suggestion
        try {
            return valueOf(operator);
        } catch (IllegalArgumentException e) {
            return null;
        }
```

`valueOf` 를 활용해보셔도 좋을 것 같아요!

Reply:
도진님의 피드백을 받아 valueOf로 수정하려 했으나,
해당 메소드의 의도는 operator를 값으로 갖는 '상수'를 return하는 것이 아니라
operator를 값으로 갖는 Operator ENUM객체를 반환하는 것이기 때문에 따로 수정을 하지 않았습니다.

Reply:
아하 그렇다면 Map으로 static 초기화시켜놓고 그걸 가져다가 쓰는 게 좋을 것 같아요!